lang/cs/Org.Apache.REEF.Client/YARN/YARNREEFClient.cs

```diff
@@ -123,42 +124,76 @@ private void Launch(IJobSubmission jobSubmission, string driverFolderPath)
 
             // TODO: Remove this when we have a generalized way to pass config to java
             var paramInjector = TangFactory.GetTang().NewInjector(jobSubmission.DriverConfigurations.ToArray());
-                
-            var avroJobSubmissionParameters = new AvroJobSubmissionParameters
+
+            var submissionJobArgsFilePath = SerializeJobFile(jobSubmission, driverFolderPath);
+            var submissionAppArgsFilePath = SerializeAppFile(jobSubmission, paramInjector, driverFolderPath);
+
+            // Submit the driver
+            _javaClientLauncher.Launch(JavaClassName, submissionJobArgsFilePath, submissionAppArgsFilePath);
+            Logger.Log(Level.Info, "Submitted the Driver for execution." + jobSubmission.JobIdentifier);
+        }
+
+        private string SerializeAppFile(IJobSubmission jobSubmission, IInjector paramInjector, string driverFolderPath)
```

Comment:
@afchung it is not just about single piece of code vs duplication. YarnREEFDotNET client should ideally have very little logic and just string calls to appropriate interfaces together to accomplish its task. In this way the code is easy to swap in and out in long term and unittests will catch most regressions. Initially it was a small logic not it seems to be growing hence I added that comment. If you feel that it is better to do this in another phase that is a reasonable decision as well.